torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py

```diff
@@ -168,83 +166,53 @@ def _annotate_linear(
-                if weight_or_bias.ndim == 2:  # type: ignore[attr-defined]
-                    weight_node = node
-                if weight_or_bias.ndim == 1:  # type: ignore[attr-defined]
-                    bias_node = node
-            if weight_node is None:
-                raise ValueError("No weight found in Linear pattern")
-            # find use of act node within the matched pattern
-            act_use_node = None
-            # When doing tracing with dynamic shape, we end up with sym_size nodes
-            # This nodes do not need quantization, so skip those.
-            # We can also have quant workflow throw exception when sym_size nodes
-            # are annotated.
-            # This is not specific to linear, so in future diffs we should streamline
-            # this.
-            weight_node_users = list(
-                filter((lambda x: (_is_sym_size_node(x) is False)), weight_node.users)
+    for node in gm.graph.nodes:
+        if node.op != "call_function" or node.target != torch.ops.aten.linear.default:
+            continue
+        if filter_fn and not filter_fn(node):
+            continue
+        act_node = node.args[0]
+        weight_node = node.args[1]
+        bias_node = None
+        if len(node.args) < 2:
+            raise ValueError(
+                f"aten.linear nodes less than 2 args which are: {node.args}"
             )
-            weight_use_node_in_p = set(weight_node_users).intersection(set(p.nodes))
-            if len(weight_use_node_in_p) != 1:
-                raise ValueError(
-                    f"Could not find a valid use of act node. All uses {weight_use_node_in_p}"
-                )
-            weight_use_node = weight_use_node_in_p.pop()
-
-            act_node_users = list(
-                filter((lambda x: (_is_sym_size_node(x) is False)), act_node.users)
+        if len(node.args) > 2:
+            bias_node = node.args[2]
+
+        if _is_annotated([node]) is False:  # type: ignore[list-item]
+            _annotate_input_qspec_map(
+                node,
+                act_node,
+                input_act_qspec,
             )
-            act_use_node_in_p = set(act_node_users).intersection(set(p.nodes))
-            if len(act_use_node_in_p) != 1:
-                raise ValueError(
-                    f"Could not find a valid use of act node. All uses {act_use_node_in_p}"
-                )
-            act_use_node = act_use_node_in_p.pop()
-            if _is_annotated([act_use_node]) is False:  # type: ignore[list-item]
-                _annotate_input_qspec_map(
-                    act_use_node,
-                    act_node,
-                    input_act_qspec,
+            _annotate_input_qspec_map(
+                node,
+                weight_node,
+                weight_qspec,
+            )
+            nodes_to_mark_annotated = [node, weight_node]
+            if bias_node and _is_annotated([bias_node]):
```

Comment:
I feel we don't need this check?

Reply:
why not?

Reply:
In fact I should check the same for weight node as well

Reply:
I think annotating bias is still valid?

Reply:
are you saying that as an edge annotation it is still valid to annotate? Yeah that makes sense, although as a parameter on linear, do we want that? this allows us associating one linear with one weight and one bias. 

What you are saying is also valid, but I am erring on the side of caution for now to not allow this behavior and introduce inadvertent behaviort

Reply:
quantizing output for bias_node won't interfere with the annotation for bias input for linear node I think since if it does not match the linear config, we'll insert a new observer.

quantizing input for bias_node is not related to the linear annotation, so I don't think it should be banned here.

I guess erroring out is fine as well since most of the time these are get_attr nodes and not shared with other ops, just feels like it's not needed. it might break in some cases but it's fine to adjust later as well.

Reply:
I think you are right. Lets just allow the use annotation. It does have side effect when there are multiple uses of weight and bias, although very unlikely, but it is ok for quantizer to not be opinionated about it.